### PR TITLE
feat: Support multi-select for "Add to Workspace" feature

### LINKED ISSUE #3324: Add to workspace using multi-select
**Is your feature request related to a problem? Please describe.**

As a user interested in setting up a workspace that contains several different PDSs, I need to right click on each PDS to add it to my workspace. 

**Describe the solution you'd like**

It would be great to be able to select multiple data sets (or USS directories) and add them all to my workspace with a single context menu action.

**Describe alternatives you've considered**

Right-clicking each data set works, but is not ideal.

**Additional context**

If you select multiple data sets, and click "Add to workspace" from the context menu, only one of the data sets you have selected is added to the workspace.

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Resolves #3324 

## Release Notes

Milestone: 3.1.0

Changelog:

- Added multi-select support for the "Add to Workspace" feature in Data Sets and USS. [#3324](https://github.com/zowe/zowe-explorer-vscode/issues/3324)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [x] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by anaxceron — COMMENTED** [review]
Left a suggestion to edit the changelog

File: packages/zowe-explorer/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 - Power users and developers can now build links to efficiently open mainframe resources in Zowe Explorer. Use the **Copy External Link** option in the context menu to get the URL for a data set or USS resource, or create a link in the format `vscode://Zowe.vscode-extension-for-zowe?<ZoweResourceUri>`. For more information on building resource URIs, see the [FileSystemProvider wiki article](https://github.com/zowe/zowe-explorer-vscode/wiki/FileSystemProvider#file-paths-vs-uris). [#3271](https://github.com/zowe/zowe-explorer-vscode/pull/3271)
 - Implemented more user-friendly error messages for API or network errors within Zowe Explorer. [#3243](https://github.com/zowe/zowe-explorer-vscode/pull/3243)
 - Use the "Troubleshoot" option for certain errors to obtain additional context, tips, and resources for how to resolve the errors. [#3243](https://github.com/zowe/zowe-explorer-vscode/pull/3243)
+- Added multi-select support for the "Add to Workspace" feature in Data Sets and USS. [#3324](https://github.com/zowe/zowe-explorer-vscode/issues/3324)
```
[the remark this answers is not in the document]
Comment (traeok) [reply packages/zowe-explorer/CHANGELOG.md]: Thanks, resolved in 417268a

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks @traeok

**Review by adam-wolfe — APPROVED** [review]
Looks good. Thanks @traeok

**Pull request merged**